### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by aborsu**

### CODE REVIEW

**Comment by SagaStalin** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [creator.py#L46](https://github.com/Sagacify/logger-py/blob/ec0d0867440ad373b3ee51c1eb259d799367c7ea/src/creator.py#L46): Remove those useless parentheses [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1110)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [formatter.py#L47](https://github.com/Sagacify/logger-py/blob/6cc2c88f2ae2ca1f83d7c7bf0814da3e37c4f460/src/formatter.py#L47): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [formatter.py#L47](https://github.com/Sagacify/logger-py/blob/6cc2c88f2ae2ca1f83d7c7bf0814da3e37c4f460/src/formatter.py#L47): This function has 7 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MINOR [__init__.py#L13](https://github.com/Sagacify/logger-py/blob/6cc2c88f2ae2ca1f83d7c7bf0814da3e37c4f460/src/__init__.py#L13): Invalid object 'SagaFormatter' in __all__, must contain only strings [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0604)
1. MINOR [__init__.py#L13](https://github.com/Sagacify/logger-py/blob/6cc2c88f2ae2ca1f83d7c7bf0814da3e37c4f460/src/__init__.py#L13): Invalid object 'get_logger' in __all__, must contain only strings [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0604)
1. MINOR [formatter.py#L27](https://github.com/Sagacify/logger-py/blob/6cc2c88f2ae2ca1f83d7c7bf0814da3e37c4f460/src/formatter.py#L27): as argument [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0102)

File: src/__init__.py [hunk 1]
```diff
@@ -3,7 +3,13 @@
...
+from .__about__ import (
+    __author__, __copyright__, __email__, __license__, __summary__,
+    __title__, __uri__, __version__
+)
 from .formatter import SagaFormatter
 from .creator import get_logger
 
-__all__ = [SagaFormatter, get_logger]
+__all__ = [
+    SagaFormatter, get_logger, __author__, __copyright__, __email__,
```
Comment (SagaStalin) [inline src/__init__.py]: MINOR Invalid object 'SagaFormatter' in __all__, must contain only strings [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0604)
MINOR Invalid object 'get_logger' in __all__, must contain only strings [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0604)
MINOR Undefined variable name 'Sagacify' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)
MINOR Undefined variable name 'dev+python@sagacify.com' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)

File: src/__init__.py [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
+    __license__, __summary__, __title__, __uri__, __version__]
```
Comment (SagaStalin) [inline src/__init__.py]: MINOR Undefined variable name 'MIT' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)
MINOR Undefined variable name 'A json log formatter.' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)
MINOR Undefined variable name 'sagalogger' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)
MINOR Undefined variable name 'https://www.github.com/Sagacify/logger-py' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)
MINOR Undefined variable name '0.3.0' in __all__ [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0603)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 12 issues
* MAJOR 2 major
* MINOR 10 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [formatter.py#L47](https://github.com/Sagacify/logger-py/blob/3ea2b84184c45d2f83ee8f97d27b1393cee9ba51/src/formatter.py#L47): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. MAJOR [formatter.py#L47](https://github.com/Sagacify/logger-py/blob/3ea2b84184c45d2f83ee8f97d27b1393cee9ba51/src/formatter.py#L47): This function has 7 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. MINOR [formatter.py#L27](https://github.com/Sagacify/logger-py/blob/3ea2b84184c45d2f83ee8f97d27b1393cee9ba51/src/formatter.py#L27): as argument [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0102)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by aborsu** [comment]
[xs:(logger-py) get name and version](https://trello.com/c/BzZWkOxP/430-xs-logger-py-get-name-and-version)

File: src/formatter.py [hunk 3]
```diff
@@ -1,199 +1,98 @@
...
-    return hasattr(key, 'startswith') and key.startswith(value)
+import time
+import json
 
 
-def merge_record_extra(record, target, reserved=[]):
-    """
-    Merges extra attributes from LogRecord object into target dictionary.
+class SagaEncoder(json.JSONEncoder):
+    def default(self, obj):
```
Comment (SagaStalin) [inline src/formatter.py]: MAJOR An attribute defined in json.encoder line 158 hides this method [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE0202)
MAJOR This function has 5 returns or yields, which is more than the 3 allowed. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 4 issues
* MAJOR 3 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/formatter.py [hunk 4]
```diff
@@ -1,199 +1,96 @@
...
-    return handler
+    https://en.wikipedia.org/wiki/ISO_8601
+    """
+    utc_time = time.gmtime(record.created)
+    time_string = time.strftime('%Y-%m-%d %H:%M:%S', utc_time)
+    return '%s.%03dZ' % (time_string, record.msecs)
 
 
-class SagaFormatter(logging.Formatter):
+class SagaFormatter(object):
```
Comment (ghost) [inline src/formatter.py:57]: MINOR Too few public methods (1/2) [rule](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AR0903)

**Comment by ghost** [comment]
SonarQube analysis reported 1 issue
- MINOR 1 minor

Watch the comments in this conversation to review them.

**Review by kfrancoi — APPROVED** [review]

File: src/formatter.py (hunk 4 above)
Comment (satblip) [reply src/formatter.py:57]: tu vas laisser un warning sonar, je ne te reconnais plus... :p

**Review by satblip — APPROVED** [review]
LGTM

**Pull request merged**